chain/chain/src/chain.rs

```diff
@@ -145,6 +145,7 @@ impl OrphanBlockPool {
 /// Chain genesis configuration.
 #[derive(Clone)]
 pub struct ChainGenesis {
+    pub chain_id: String,
```

Comment:
hm, a bit weird to have chain_id in ChainGenesis.
You can pass it to NetworkConfig directly instead of routing via chain.

Reply:
I took it out, let me know if it makes more sense now.